pkg/auth/setup.go

```diff
@@ -47,6 +52,85 @@ func attachPolicies(ctx context.Context, authService Service, groupID string, po
 	return nil
 }
 
+// statementForPolicyType holds the Statement for a policy by its name,
+// without the required ARN.
+var statementByName = map[string]model.Statement{
+	"AllAccess": {
+		Action: []string{"fs:*", "auth:*", "ci:*", "retention:*", "branches:*"},
+		Effect: model.StatementEffectAllow,
+	},
+	"FSFullAccess": {
+		Action: []string{
+			"fs:*",
+		},
+		Effect: model.StatementEffectAllow,
+	},
+	"FSReadWrite": {
+		Action: []string{
+			permissions.ListRepositoriesAction,
+			permissions.ReadRepositoryAction,
+			permissions.ReadCommitAction,
+			permissions.ListBranchesAction,
+			permissions.ListTagsAction,
+			permissions.ListObjectsAction,
+			permissions.ReadObjectAction,
+			permissions.WriteObjectAction,
+			permissions.DeleteObjectAction,
+			permissions.RevertBranchAction,
+			permissions.ReadBranchAction,
+			permissions.ReadTagAction,
+			permissions.CreateBranchAction,
+			permissions.CreateTagAction,
+			permissions.DeleteBranchAction,
+			permissions.DeleteTagAction,
+			permissions.CreateCommitAction,
+		},
+		Effect: model.StatementEffectAllow,
+	},
+	"FSRead": {
+		Action: []string{
+			"fs:List*",
+			"fs:Read*",
+		},
+
+		Effect: model.StatementEffectAllow,
+	},
+	"AuthManageOwnCredentials": {
+		Action: []string{
+			permissions.CreateCredentialsAction,
+			permissions.DeleteCredentialsAction,
+			permissions.ListCredentialsAction,
+			permissions.ReadCredentialsAction,
+		},
+		Effect: model.StatementEffectAllow,
+	},
+}
+
+// MakeStatementForPolicyType returns statements for policy type typ,
+// limited to resources.
+func MakeStatementForPolicyType(typ string, resources []string) (model.Statements, error) {
+	statement, ok := statementByName[typ]
+	if !ok {
+		return nil, fmt.Errorf("%w: %s", ErrStatementNotFound, typ)
+	}
+	statements := make(model.Statements, len(resources))
+	for i, resource := range resources {
+		if statements[i].Resource == "" {
```

Comment:
isn't this `if` redundant?

Reply:
True.  It should look at statement, which _might_ in future be configured with a specific resource.  It was meant as a safety measure for when that day comes, so we do not end up _widening_ a limited statement in `statementByName`.